functions/Get-DbaAgentSchedule.ps1

```diff
@@ -249,18 +258,26 @@ function Get-DbaAgentSchedule {
                 Stop-Function -Message "$($server.Edition) does not support SQL Server Agent. Skipping $server." -Continue
             }
 
-            if ($Schedule) {
-                $scheduleCollection = $server.JobServer.SharedSchedules | Where-Object { $_.Name -in $Schedule }
+            $scheduleCollection = @()
+
+            if ($Schedule -or $ScheduleUid -or $Id) {
+                if ($Schedule) {
+                    $scheduleCollection += $server.JobServer.SharedSchedules | Where-Object { $_.Name -in $Schedule }
+                }
+
+                if ($ScheduleUid) {
+                    $scheduleCollection += $server.JobServer.SharedSchedules | Where-Object { $_.ScheduleUid -in $ScheduleUid }
+                }
+
+                if ($Id) {
+                    $scheduleCollection += $scheduleCollection | Where-Object { $_.Id -in $Id }
```

Comment:
If only $Id is given (and not $Schedule or $ScheduleUid) $scheduleCollection is still empty here.

Reply:
You're right, I overlooked that one

Reply:
![image](https://user-images.githubusercontent.com/6154981/86001713-92e55400-ba0f-11ea-9ca9-f7688c80b7d0.png)